### ISSUE TITLE
Interrupts already defined

### ISSUE THREAD

**Issue by MatthiasW1004** [MOSAIC-LoPoW/octa-stack-students#55]
If I try to generate an interrupt using a push button, I can't make the bin file because the EXTI0_IRQHandler is already defined elsewhere. How can I make it use my IRQHandler? 
![image](https://user-images.githubusercontent.com/63510088/150998782-a4f5d0a2-79df-4c5e-a1e4-b5c6ad9aa5a5.png)
![image](https://user-images.githubusercontent.com/63510088/150998809-7567650d-9c35-4af5-8a8a-e8d35d823209.png)

**Comment by MatsDeMeyer** [comment]
The clean solution is using the EXTI0_IRQHandler that is already defined in the stm32l4xx_it file, and placing your code there.
If you really want to use your own handler, delete the already defined one or make it "weak" so that it uses another one if there is one

**Comment by MatthiasW1004** [comment]
I fixed it like this:
![image](https://user-images.githubusercontent.com/63510088/151016138-b0dbcedb-80ca-426b-b118-fda4d4f86005.png)

**Comment by MatthiasW1004** [comment]
For the RTC alarm callback, I want to use HAL_RTC_RegisterCallback. It is defined in the stm32l4xx_hal_rtc file but it does not recognize the defined parameters. 
![image](https://user-images.githubusercontent.com/63510088/151026021-c84befc0-102e-4753-adb2-50cfc5339d30.png)
![image](https://user-images.githubusercontent.com/63510088/151026067-109151f3-1657-4ea2-a3df-6b6ec24aa1e2.png)

**Comment by MatsDeMeyer** [comment]
VSCode does tend to highlight stuff as undefined, does your code compile?

**Comment by MatthiasW1004** [comment]
No it does not, it gives me the 'not declared' error

**Comment by Klievan** [comment]
Could you post the full error the compiler is giving?

**Comment by MatthiasW1004** [comment]
![image](https://user-images.githubusercontent.com/63510088/151146839-e9855194-a77c-4e15-8b60-3d111f06b6f4.png)

**Comment by MatthiasW1004** [comment]
I am also wondering why the code where this function is defined is more blurry than the code above: 
![image](https://user-images.githubusercontent.com/63510088/151148291-9e84cf10-1cc8-449c-a76b-36fbe977624b.png)

**Comment by MatthiasW1004** [comment]
Do you maybe have an example in your octastack about how to use the RTC-Alarm interrupt? Or something similar?

**Comment by Klievan** [comment]
Hi Matthias,

Mats is unavailable right now but I'll try to help you out here. Like you noticed, the function that you're trying to call is dimmed, which means that the #if directive above is probably returning false, so the code below is not being included. Try defining USE_HAL_RTC_REGISTER_CALLBACKS to 1 in your makefile
https://stackoverflow.com/questions/53288150/how-to-add-define-in-makefile

**Comment by MatthiasW1004** [comment]
I  changed the value in the hal_conf_template file but it still does not work. For some reason I also cannot add the define in the make file as said in the stackoverflow discussion

**Comment by Klievan** [comment]
Can you post the line you added in your makefile?

**Comment by Klievan** [comment]
Are you getting the same error?

**Comment by MatthiasW1004** [comment]
![image](https://user-images.githubusercontent.com/63510088/151169326-8d1d12d1-d381-4fc9-9586-ede731c22f59.png)

And I get the same error. I also tried changing it here: 
![image](https://user-images.githubusercontent.com/63510088/151167044-daebdf34-f213-410f-95a0-f82fd35c434c.png)

**Comment by MatthiasW1004** [comment]
![image](https://user-images.githubusercontent.com/63510088/151167291-3a28f4ce-feb9-4977-bcd5-63d5a76feb44.png)
This is the documentation about it

**Comment by Klievan** [comment]
And the error remains the same? Have you tried to do a clean build?

**Comment by MatthiasW1004** [comment]
Yes, tried the clean build, even tried other ways to define the USE_HAL_RTC_REGISTER_CALLBACKS but still the same error.

**Comment by Klievan** [comment]
Did you import ./core/ST/STM32L4xx_HAL_Driver/Inc/stm32l4xx_hal_rtc.h?

**Comment by MatthiasW1004** [comment]
![image](https://user-images.githubusercontent.com/63510088/151183169-2896f93e-07d8-4060-b773-48085168ff8d.png)
I did like this, I think it is correct right?

**Comment by Klievan** [comment]
It is if you also added the correct entries in your makefile. Could you show those?

**Comment by MatthiasW1004** [comment]
![image](https://user-images.githubusercontent.com/63510088/151185397-a519d72c-f518-4590-bd33-58db0c438ecd.png)

This is the full make file
